diesel_tests/tests/types_roundtrip.rs

```diff
@@ -344,6 +348,64 @@ pub fn mk_naive_date(days: u32) -> NaiveDate {
     earliest_sqlite_date + Duration::days(days as i64 % num_days_representable)
 }
 
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+#[derive(Clone, Debug)]
+struct SerdeWrapper(serde_json::Value);
+
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+impl quickcheck::Arbitrary for SerdeWrapper {
+    fn arbitrary<G: quickcheck::Gen>(g: &mut G) -> Self {
+        SerdeWrapper(arbitrary_serde(g, 0))
+    }
+}
+
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+fn arbitrary_serde<G: quickcheck::Gen>(g: &mut G, depth: usize) -> serde_json::Value {
+    use rand::distributions::Alphanumeric;
+    use rand::Rng;
+    let variant = match g.gen_range(0, 6) {
```

Comment:
Couldn't this whole thing be written as `g.gen_range(0, if depth > 0 { 6 } else { 4 })`?